src/main/haskell/kore/app/exec/Main.hs

```diff
@@ -334,17 +358,75 @@ main = do
                                         (fst simplifiedPatterns) of
                                     [] -> ExpandedPattern.bottom
                                     (config : _) -> config
-                        pickLongest <$> runStrategy
-                            (transitionRule
-                                metadataTools
-                                substitutionSimplifier
-                                simplifier
-                            )
-                            (strategy axiomPatterns)
-                            stepLimit
-                            (initialPattern, mempty)
+                        give (symbolOrAliasSorts metadataTools)
+                            ExpandedPattern.toMLPattern . fst . pickLongest
+                            <$> computeStrategy initialPattern
+                Just sType -> do
+                    searchPattern <-
+                        mainParseSearchPattern indexedModule metadataTools
+                            searchFileName
+                    clockSomething "Searching"
+                        $ evalSimplifierWithTimeout smtTimeOut
+                        $ do
+                            axiomsAndSimplifiers <-
+                                makeAxiomsAndSimplifiers indexedModule metadataTools
+                            let
+                                (rewriteAxioms, simplifier, substitutionSimplifier) =
+                                    axiomsAndSimplifiers
+                                runStrategy' pat =
```

Comment:
Really nice job refactoring!  But, while you're at it, the definition of `runstrategy'` here seems identical to that of `computeStrategy` above.  Do you think it would make sense to extract it?